Eatery/Views/Eateries/HistogramViewController.swift

```diff
@@ -0,0 +1,285 @@
+        setUpAxis()
+    }
+    
+    // MARK: Initial view set up
+    
+    private func setUpAxisLineView() {
+        let axisView = UIView()
+        axisView.backgroundColor = .inactive
+        view.addSubview(axisView)
+        
+        axisView.snp.makeConstraints { make in
+            make.height.equalTo(2)
+            make.bottom.equalToSuperview().inset(45)
+            make.leading.equalToSuperview()
+            make.trailing.equalToSuperview().inset(15);
+        }
+    }
+    
+    private func setUpAxis() {
+        setUpAxisLineView()
+        
+        let overflowingEndTime = (hourRange.endTime < hourRange.startTime) ? hourRange.endTime + 24 : hourRange.endTime
+        let tickCount: Int = ((overflowingEndTime - hourRange.startTime) / 3) + 1
+        
+        var x: Double = 0
+        for tick in 0..<tickCount {
+            let tickView = UIView()
+            axisTickViews.append(tickView)
+            view.addSubview(tickView)
+            
+            tickView.backgroundColor = .inactive
+            tickView.snp.makeConstraints { make in
+                make.bottom.equalTo(view).inset(35)
+                make.leading.equalTo(view).offset(x)
+                make.width.equalTo(2)
+                make.height.equalTo(10)
+            }
+            
+            let tickLabel = UILabel()
+            let tickHour = containOverflowedMilitaryHour(hour: hourRange.startTime + (tick * 3))
+            tickLabel.text = formattedHourForTime(militaryHour: tickHour)
+            tickLabel.textColor = .secondary
+            view.addSubview(tickLabel)
+            
+            tickLabel.snp.makeConstraints { make in
+                make.bottom.equalTo(view).inset(10)
+                make.leading.equalTo(view).offset(x)
+                make.width.equalTo(30)
+                make.height.equalTo(15)
+            }
+            
+            let additionalSpacing = !(tick == 0 || tick >= (tickCount - 2))
+            let xSpacingAdd: Double = (((additionalSpacing ? 2 : 0) + 1) * 2) + ((tick >= (tickCount - 2)) ? 4 : 0)
+            x += (3 * barWidth) + xSpacingAdd
+        }
+    }
+    
+    private func setUpBarContainerView() {
+        barContainerView = UIView()
+        view.addSubview(barContainerView)
+        
+        barContainerView.snp.makeConstraints { make in
+            make.leading.equalTo(view)
+            make.bottom.equalTo(view).inset(47)
+            make.width.equalTo(view)
+            make.height.equalTo(view).inset(65)
+        }
+    }
+    
+    private func setUpBarViews() {
+        setUpBarContainerView()
+        
+        let barCount = (overflowingEndTime - hourRange.startTime) - 2
+        let frameWidth = Double(view.frame.width) - 15
+        let spacingLoss = Double(2 * (barCount - 1))
+        let maxBarHeight = Double(view.frame.height - 102) // 102 is the net space needed for other histogram components
+        barWidth = (frameWidth - spacingLoss) / Double((barCount + 2))
```

Comment:
More math and constants we have to preserve... We should aim to use constraints to remove these constants as well.